### ISSUE TITLE
Data importer keeps returning "This access request is invalid. Please never follow this link again.", but only on second user account

### ISSUE THREAD

**Issue by alexklibisz** [firefly-iii/firefly-iii#5576]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I'm running firefly on https://firefly.mydomain.com and the new data-importer on https://import.firefly.mydomain.com, with the following env vars:

```yaml
      - TRUSTED_PROXIES=**
      - FIREFLY_III_URL=https://firefly.mydomain.com
      - FIREFLY_III_CLIENT_ID=499
      - JSON_CONFIGURATION_DIR=/mnt/configurations
```

I can authenticate in the importer and import data using my main firefly account. This is also the account that I used to create the client referenced in the client_id env var. 

However, I also have a second account on the same firefly instance, and when I try to use the importer from this second account, I keep getting this error:

![image](https://user-images.githubusercontent.com/8015228/150270111-df879ea7-3639-4562-ae28-bf4b1fc0ecf5.png)

### Expected behaviour

Data-importer should work on main account and second account.

### Debug information

Firefly /debug

Debug information generated at 2022-01-19 22:48:13 America/New_York for Firefly III version **5.6.10**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.6.10 |
| Firefly III API | 1.5.5 |
| Build | 75, 09-01-2022 08:36:35 CET |
| PHP | 8.0.13 |
| Host | Linux |

| System info | Value |
| --- | --- |
| Installation ID | e1cc6dba-0dbc-4d29-b747-a26a86557361 |
| Using docker? | true |
| Layout | v1 |
| System TZ | America/New_York |
| Browser TZ | America/New_York |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | info, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| Default language | en_US |
| Default locale | equal |
| BCscale | 24 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | pgsql |
| DB version | 18 (exp. 18) |
| Login provider | eloquent  |
| Trusted proxies (.env) | ** |

| User info | Value |
| --- | --- |
| Session start | 2022-01-01 00:00:00 |
| Session end | 2022-12-31 23:59:59 |
| Session first | 2019-03-01 00:00:00 |
| User ID | 1 |
| User language | en_US |
| User locale | en_US |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| User agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.71 Safari/537.36 |

data-importer /debug

Debug information generated at 2022-01-20 04:48:33 Europe/Amsterdam for Firefly III Data Importer version **0.6.5**.

| Scope | Version |
| --- | --- |
| FIDI | 0.6.5 |
| Build | 76, 14-01-2022 20:50:11 CET |
| PHP | 8.0.13 |
| Host | Linux |

| System info | Value |
| --- | --- |
| Using docker? | true |
| System TZ |  |
| Browser TZ | America/New_York |
| App environment | production |
| App debug mode | false |
| App cache driver | file |
| App logging | debug, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| BCscale | 12 |
| Trusted proxies (.env) | ** |

| User info | Value |
| --- | --- |
| User agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.71 Safari/537.36 |

### Additional info

This has also happened before with the csv-importer, when I tried to upgrade to firefly 5.6.4 and csv-importer 2.6.1. I ended up reverting to 5.6.3 and 2.6.0. But I also ran firefly and the csv-importer successfully with 5.6.8 and 2.6.1. Then today it started failing again on 5.6.10 and 2.6.1, which is when I decided to go ahead and try the data-importer, but that has the same issue, as described here. So it seems like it could be some transient issue.

I've tried a few things to no avail:
- Re-created the OAuth client about a dozen times.
- Try a different browser
- Clear browser cookies and local storage
- Re-create the OAuth client

Here's the URL of that error page: `https://firefly.mydomain.com/oauth/authorize?client_id=499&code_challenge=XP9RqGSZQmag-Modo1ZiV9plJDYcYD8CHepwX8joVC4&code_challenge_method=S256&redirect_uri=https%3A%2F%2Fimport.firefly.mydomain.com%2Fcallback&response_type=code&scope=&state=8LMLAxM0j2MNuTch8kXe9rtGPOfKo0Rg2mnXRE6t`

My Oauth client seems fine, right?

![image](https://user-images.githubusercontent.com/8015228/150270701-c917dcc2-624f-4c8f-84d0-8946d5901a51.png)

**Comment by JC5** [comment]
It looks fine. The error happens when you try to authenticate with another user's oAuth record.

I'll see if I can replicate it.

**Comment by JC5** [comment]
The fact that you hard-coded the `FIREFLY_III_CLIENT_ID` maybe an issue, when it's combined with you being logged in with another user.

Does it work when you remove the `FIREFLY_III_CLIENT_ID` setting?

**Comment by alexklibisz** [comment]
Hey James, thanks for the quick response. So based on your note, I got it working with this workaround:

1. Remove the FIREFLY_III_CLIENT_ID env var
2. Create a new Oauth client from the second account
3. Use that Oauth client ID to authenticate the second account

It's a bit roundabout, but I guess I'm unblocked. I'm curious how this worked for so long with a single OAuth client. Like I said above, I never had issues before Firefly 5.6.4 and csv-importer 2.6.1. Maybe that's a hint for where to look for the root cause?

**Comment by JC5** [comment]
It was a bug, actually. Although there is no direct security risk, the client ID has to match. So this will not be "fixed", unfortunately.

**Comment by alexklibisz** [comment]
That's fine. Just making sure I understand, are you saying each user has to have a distinct OAuth client?

**Comment by JC5** [comment]
Yes, that's the case indeed. It means they can individually revoke them as well.

**Comment by alexklibisz** [comment]
Ok, good to know. I guess I had expected that the admin account could own a single OAuth client to be shared by multiple users. Anyway, it's working now. Thanks!
